app/main/views/direct_award.py

```diff
@@ -348,3 +343,71 @@ def record_project_download(project_external_id):
     db.session.commit()
 
     return single_result_response("project", project), 200
+
+
+@main.route('/direct-award/projects/<int:project_external_id>/services/<string:service_id>/award', methods=['POST'])
+def create_award_process_outcome(project_external_id, service_id):
+    updater_json = validate_and_return_updater_request()
+
+    # Acquire share-lock on project and active search rows - what we're trying to assert here is that the "locked" field
+    # of the project and the "active" field of the search are set to appropriate values at the time we create this
+    # process outcome
+    project = db.session.query(DirectAwardProject).options(
+        # this has to be performed as an inner join for the row-lock to work, just means we have to perform an extra
+        # database hit in the no-rows case to disambiguate the error message
+        db.joinedload(DirectAwardProject.active_search, innerjoin=True)
+    ).filter(
+        DirectAwardProject.external_id == project_external_id,
+    ).with_for_update(read=True).first()
+
+    if project is None:
+        if db.session.query(DirectAwardProject.id).filter_by(external_id=project_external_id).first():
+            # by deduction...
```

Comment:
This comment is not helpful

Reply:
This relates and can be seen as a continuation of the previous comments. It doesn't make sense on its own, but I think it's okay in context.

Reply:
I will expand it though